src/oc/hoc.c

```diff
@@ -745,6 +749,52 @@ void hoc_coredump_on_error(void) {
 	pushx(1.);
 }
 
+void print_bt() {
+#ifdef USE_BACKWARD
+    backward_wrapper();
+#else
+#if HAVE_EXECINFO_H
+    const size_t nframes = 12;
+    void *frames[nframes];
+    size_t size;
+    char** bt_strings = NULL;
+    size_t funcname_size = 256;
+    char* symbol = malloc(sizeof(char)*funcname_size);
+    char* offset = malloc(sizeof(char)*10);
```

Comment:
I assume these are not freed but as program is terminating we are not worrying?

A bit related question - why this function was implemented in .cpp file somewhere with `extern` so that we don't have to worry about all C memory management?

Reply:
I forgot to free them, my mistake!
goot point, regarding C++ implementation.. I should maybe do that.

Reply:
Upon further consideration I decided not to do that. This code deals with mostly C API calls to do the demangling, retrieval of the call stack and printing, so there is little value in writing it in C++ because you would still need to pass C style buffers to those functions.